server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/KeywordsConsistencyOnDelegationMailbox.feature

```diff
@@ -0,0 +1,190 @@
+# or more contributor license agreements.  See the NOTICE file *
+# distributed with this work for additional information        *
+# regarding copyright ownership.  The ASF licenses this file   *
+# to you under the Apache License, Version 2.0 (the            *
+# "License"); you may not use this file except in compliance   *
+# with the License.  You may obtain a copy of the License at   *
+#                                                              *
+#   http://www.apache.org/licenses/LICENSE-2.0                 *
+#                                                              *
+# Unless required by applicable law or agreed to in writing,   *
+# software distributed under the License is distributed on an  *
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+# KIND, either express or implied.  See the License for the    *
+# specific language governing permissions and limitations      *
+# under the License.                                           *
+# **************************************************************/
+
+Feature: JMAP keywords consistency on delegation mailbox
+
+  Background:
+    Given a domain named "domain.tld"
+    And a user "alice@domain.tld"
+    And a user "bob@domain.tld"
+    And "alice@domain.tld" has a mailbox "notShared"
+    And "alice@domain.tld" has a mailbox "shared"
+    And "alice@domain.tld" has a mailbox "trash"
+    And "alice@domain.tld" shares its mailbox with write right "shared" with "bob@domain.tld"
+    And "alice@domain.tld" is connected
+    And the user has a message "mess" in "notShared" mailbox with subject "My awesome subject", content "This is the content"
+    And the user copy "mess" from mailbox "notShared" to mailbox "shared"
+
+  Scenario: getMessageList filtered by flag should combine flag when delegation mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags on "mess" to "$Flagged"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailboxes "shared,notShared" with flag "$Flagged"
+    Then the message list has size 1
+    And the message list contains "mess"
+
+  Scenario: getMessageList filtered by flag should keep flag on delegation mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags on "mess" to "$Flagged"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailboxes "notShared" with flag "$Flagged"
+    Then the message list is empty
+
+  Scenario: getMessageList filtered by flag should keep flag on non-shared mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags on "mess" to "$Flagged"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailboxes "shared" with flag "$Flagged"
+    Then the message list has size 1
+    And the message list contains "mess"
+
+  @Ignore
+  Scenario: (Does not work) Get message list should insection Draft flag on all mailbox
```

Comment:
It does not work now because we search by $Draft flag first then we filter by "inMailboxes" so I don't know where we can combine/intersect flag.